hnn_core/opt_toy_example/general.py

```diff
@@ -101,270 +107,241 @@ def plot_convergence(self, ax=None, show=True):
+    predicted_params : list
+        Parameters selected by the optimizer.
 
     Returns
     -------
-    params : dictionary
-        Contains parameter names, initial parameters, and constraints.
+    params_dict : dict
+        Keys are parameter names, values are parameters.
+    """
+
+    param_dict = dict()
+    for param_key, param_name in enumerate(initial_params):
+        param_dict.update({param_name: predicted_params[param_key]})
+
+    return param_dict
 
+
+def _run_opt_bayesian(net, constraints, initial_params, set_params, obj_fun,
+                      scale_factor, smooth_window_len, tstop, max_iter,
+                      target_statistic):
     """
+    Runs optimization routine with gp_minimize optimizer.
 
-    # get initial params
-    params = _get_params(net, constraints)
+    Parameters
+    ----------
+    net : Network
+        The network object.
+    constraints : list of tuples
+        Parameter constraints in solver-specific format.
+    initial_params : dict
+        Keys are parameter names, values are initial parameters..
+    set_params : func
+        User-defined function that sets parameters in network drives.
+    obj_fun : func
+        The objective function.
+    scale_factor : float
+        The dipole scale factor.
+    smooth_window_len : float
+        The smooth window length.
+    tstop : float
+        The simulated dipole's duration.
+    max_iter : int
+        Number of calls the optimizer makes.
+    target_statistic : ndarray
+        The recorded dipole.
 
-    # assemble constraints in solver-specific format
-    cons_cobyla = list()
-    for cons_idx, cons_val in enumerate(params['constraints']):
-        cons_cobyla.append(lambda x:
-                           params['constraints'][cons_idx][1] - x[cons_idx])
-        cons_cobyla.append(lambda x:
-                           x[cons_idx] - params['constraints'][cons_idx][0])
-    params.update({'constraints': cons_cobyla})
-    return params
-
-
-def _run_opt_bayesian(net, params, obj_fun, target_statistic, max_iter,
-                      f_bands, weights):
-    """Uses gp_minimize optimizer.
-
-       Parameters
-       ----------
-       net : Network
-       params : dictionary
-           Contains parameter names, initial parameters, and constraints.
-       obj_fun : func
-           The objective function.
-        target_statistic : ndarray
-            The recorded dipole.
-        max_iter : int
-            Max number of calls.
-
-       Returns
-       -------
-       opt_params : list
-           Final parameters.
-       obj : list
-           Objective values.
-       net_ : Network
-           Optimized network object.
+    Returns
+    -------
+    opt_params : list
+        Optimized parameters.
+    obj : list
+        Objective values.
+    net_ : Network
+        Optimized network object.
     """
 
+    obj_values = list()
+
     def _obj_func(predicted_params):
```

Comment:
An alternative way to write this would be with a [partial function](https://www.geeksforgeeks.org/partial-functions-python/l)

Reply:
I was thinking the same but this might be actually cleaner and more explicit as the argument that remains to be defined (predicted_params) is clear here